SharedExpensesApp/app/src/main/java/com/example/sharedexpensesapp/ui/screens/GroupScreen.kt

```diff
@@ -1,38 +1,172 @@
 ) {
+    Box(modifier = Modifier.fillMaxSize()) {
+        Image(
+            painter = painterResource(R.drawable.illustration_sans_titre_35),
+            contentDescription = "image de fond",
+            contentScale = ContentScale.Crop,
+            modifier = Modifier
+                .matchParentSize()
+        )
+    }
     Column(
         modifier = modifier,
-        horizontalAlignment = Alignment.CenterHorizontally,
-        verticalArrangement = Arrangement.Center
     ) {
-        Button(
-            onClick = onAddExpenseButtonClicked,
-            colors = ButtonDefaults.buttonColors(
-                containerColor = MaterialTheme.colorScheme.primary,
-                contentColor = MaterialTheme.colorScheme.onPrimary
-            ),
+        ExpenseBar(selectedGroup = selectedGroup, modifier = modifier.fillMaxWidth())
+        Spacer(modifier = Modifier.weight(1f))
+        Row {
+            Spacer(modifier = Modifier.weight(1f))
+            Button(
+                onClick = onAddExpenseButtonClicked,
+                colors = ButtonDefaults.buttonColors(
+                    containerColor = colorResource(R.color.main_orange),
+                    contentColor = Color.White),
+                modifier = Modifier
+                    .padding(10.dp)
+
+            ) {
+                Text(
+                    text = "Add new",
+                    fontSize = 20.sp,
+                    modifier = Modifier
+                        .padding(10.dp)
+                )
+            }
+        }
+    }
+}
+
+@Composable
+fun ExpenseBar(
+    selectedGroup: GroupItem?,
+    modifier: Modifier = Modifier
+) {
+    Column(
+        modifier = Modifier
+            .fillMaxWidth()
+            .background(colorResource(R.color.main_purple)),
+    ) {
+        Row(
+            modifier = Modifier
+                .padding(start = 15.dp, end = 15.dp, bottom = 5.dp, top = 10.dp)
+        ) {
+            Text(
+                text = "Total expense",
+                color = Color.White,
+                fontSize = 25.sp
+            )
+            Spacer(modifier = Modifier.weight(1f))
+            Text(
+                text = selectedGroup?.balance.toString() + selectedGroup?.currency,
+                color = Color.White,
+                fontSize = 25.sp
+            )
+        }
+        Row(
             modifier = Modifier
-                .padding(start = 10.dp, end = 10.dp)
+                .padding(start = 15.dp, end = 15.dp, bottom = 15.dp, top = 3.dp)
         ) {
-            Text(text = "Add expense", fontSize = 20.sp)
+            Text(
+                text = "Spent by you",
+                color = Color.White
+            )
+            Spacer(modifier = Modifier.weight(1f))
+            Text(
+                text = selectedGroup?.balance.toString() + selectedGroup?.currency,
+                color = Color.White
+            )
         }
     }
+    selectedGroup?.expenses?.forEach { expenseItem ->
```

Comment:
Should not be Lazy List? What is going to happen if we have too many expenses?

Reply:
modified everything it should work well now